project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/services/ProjectOpeningService.kt

```diff
@@ -0,0 +1,41 @@
+package org.plan.research.minimization.plugin.services
+
+import com.intellij.ide.impl.OpenProjectTask
+import com.intellij.ide.impl.ProjectUtil
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.externalSystem.importing.ImportSpecBuilder
+import com.intellij.openapi.externalSystem.service.execution.ProgressExecutionMode
+import com.intellij.openapi.externalSystem.util.ExternalSystemUtil
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.project.guessProjectDir
+import org.jetbrains.kotlin.idea.configuration.GRADLE_SYSTEM_ID
+import java.nio.file.Path
+
+@Service(Service.Level.APP)
+class ProjectOpeningService {
```

Comment:
Do we want some flexibility in the future with it? I would like to suggest using the [heavy service](https://plugins.jetbrains.com/docs/intellij/plugin-services.html#declaring-a-service) to support mocking / future implementations

Reply:
Yes, that makes sense, but that's more about [JBRes-1977](https://youtrack.jetbrains.com/issue/JBRes-1977)